### PR TITLE
Fix two edge case bugs in MQTT and models

### PR DESCRIPTION
**Pull request by markus-lassfolk**
- Fix YarboCommandResult.from_dict to handle None state values gracefully by using 'int(d.get("state") or 0)' instead of 'int(d.get("state", 0))' to prevent TypeError when state key exists with None value

- Fix shared mutable envelope dict issue in MqttTransport._on_message by creating a separate envelope dict for each queue consumer instead of sharing the same reference, preventing data corruption across concurrent consumers

## Summary

Closes #

## Type of change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Refactor / code cleanup (no behaviour change)
- [ ] 📝 Documentation update
- [ ] ⚙️ CI/CD / tooling change

## Changes made

-
-

## Testing

- [ ] Added / updated pytest tests
- [ ] All tests pass locally (`pytest tests/`)
- [ ] Tested manually against Yarbo hardware (if applicable — describe setup below)

## Quality checklist

- [ ] **ruff lint**: Zero errors (`ruff check src/ tests/`)
- [ ] **ruff format**: No formatting issues (`ruff format --check src/ tests/`)
- [ ] **mypy**: Zero new type errors (`mypy src/yarbo/`)
- [ ] **Tests pass**: `pytest tests/` — all green
- [ ] **CHANGELOG.md** updated under `[Unreleased]`
- [ ] **Docs updated** (docstrings, README if API changed)
- [ ] **No secrets, credentials, IP addresses, or serial numbers** in code, comments, or commits
- [ ] **Follows coding standards** in [CONTRIBUTING.md](../CONTRIBUTING.md)

## Screenshots / output (optional)

---

> **Reviewer note**: Branch must be up-to-date with `develop` and all CI checks must pass before merge.

---

> [!NOTE]
> **Medium Risk**
> Touches core MQTT message fan-out by deep-copying payloads per consumer, which could impact performance or memory under high telemetry rates, but is a targeted bug fix. Model parsing change is small and covered by new tests.
> 
> **Overview**
> Fixes two edge-case bugs: `YarboCommandResult.from_dict` now treats `state=None` as `0` instead of raising, and `MqttTransport._on_message` now deep-copies the decoded payload per queue so consumers can’t mutate each other’s envelopes.
> 
> Adds/adjusts tests to cover `state=None` handling and verify envelope/payload isolation across multiple registered queues, plus minor test import cleanups and lint/type-ignore annotations in `error_reporting`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8141c05d3cfc5dd52b11518a3298b9aa597ac156. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/yarbo/mqtt.py [hunk 1]
```diff
@@ -350,8 +350,8 @@ def _on_message(self, client: Any, userdata: Any, msg: Any) -> None:
                 str(payload)[:160],
             )
             if self._loop and self._message_queues:
-                envelope: dict[str, Any] = {"topic": msg.topic, "payload": payload}
                 for q in list(self._message_queues):
+                    envelope: dict[str, Any] = {"topic": msg.topic, "payload": payload}
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply src/yarbo/mqtt.py]: Tests added in `test_mqtt.py::TestOnMessage::test_envelope_dict_isolation`: asserts that `env1 is not env2`, `env1["payload"] is not env2["payload"]`, and that mutating one envelope's payload does not affect the other. The `copy.deepcopy` in `_on_message` ensures each consumer gets an independent dict.
